cpp/src/arrow/memory_pool.h

```diff
@@ -65,6 +65,8 @@ class MemoryPoolStats {
 /// take care of the required 64-byte alignment.
 class ARROW_EXPORT MemoryPool {
  public:
+  static constexpr int64_t kDefaultAlignment = 64;
```

Comment:
It's a bit of a pity to have to include `memory_pool.h` just for this, can we instead make it a global constant (e.g. `kDefaultMemoryAlignment`) and move it to `type_fwd.h`?

Reply:
Moved it to `type_fwd.h`